src/components/SpotifyPlayer.vue

```diff
@@ -157,12 +150,8 @@
                 const deviceId = this.player._options.id
                 let state = this.currentState && this.currentState.shuffle ? 'false' : 'true'
 
-                axios.put(`https://api.spotify.com/v1/me/player/shuffle?device_id=${deviceId}&state=${state}`, {}, {
-                    headers: {
-                        'Content-Type': 'application/json',
-                        'Authorization': `Bearer ${this.token.token}`
-                    }
-                });
+                SpotifyService.shuffle(deviceId, state, this.token.token)
+                    .catch(console.error) // eslint-disable-line no-console
```

Comment:
This will need to be more robust. The main cases here are:

1. No internet (status code 0?); probably show a warning/error
2. Expired token (status code 401); open a popup, do the auth flow
3. Spotify API is out (status code >= 400) ; maybe same flow as 1
4. We introduced a bug (exception); maybe same flow as 1

We could put that logic in a service module, since there may be a number of places where we need to handle API errors.